src/test/java/org/broadinstitute/barclay/help/DefaultDocWorkUnitHandlerUnitTest.java

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.barclay.help;
+
+import com.sun.javadoc.ClassDoc;
+import org.broadinstitute.barclay.argparser.TestProgramGroup;
+import org.broadinstitute.barclay.help.testinputs.TestArgumentContainer;
+import org.broadinstitute.barclay.help.testinputs.TestExtraDocs;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class DefaultDocWorkUnitHandlerUnitTest {
+
+    @DataProvider
+    public static Object[][] workUnitDescriptions() {
+        return new Object[][] {
+                {TestArgumentContainer.class, "", Collections.EMPTY_MAP, ""},
+                {TestExtraDocs.class, "Javadoc description", Collections.EMPTY_MAP, ""},
+                {TestExtraDocs.class, "Javadoc description \nin two lines", Collections.singletonMap("", "Inline tag"), "Inline tag"}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] workUnitSummary() {
+        return new Object[][] {
+                {TestArgumentContainer.class, "", Collections.EMPTY_MAP, TestArgumentContainer.ONE_LINE_SUMMARY},
+                {TestExtraDocs.class, "Javadoc description", Collections.EMPTY_MAP, "Javadoc description"},
+                {TestExtraDocs.class, "Javadoc description \nin two lines", Collections.singletonMap("", "Inline tag"), "Javadoc description in two lines"}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] workUnitGroupName() {
+        return new Object[][] {
+                {TestArgumentContainer.class, TestArgumentContainer.GROUP_NAME},
+                {TestExtraDocs.class, TestExtraDocs.GROUP_NAME}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] workUnitGroupSummary() {
+        return new Object[][] {
+                {TestArgumentContainer.class, TestProgramGroup.DESCRIPTION},
+                {TestExtraDocs.class, Collections.EMPTY_MAP, ""},
```

Comment:
I know you're still in the middle of this, but one has two params and one has three.

Reply:
My bad, a refactoring error. Maybe we should add a test for data providers as in HTSJDK/Picard/GATK (in a different PR).